# HG changeset patch
# User Max Brister <max@2bass.com>
# Date 1342799416 18000
#      Fri Jul 20 10:50:16 2012 -0500
# Node ID 06f2e9bdade63262c266480e59b4f76a79b8b45d
# Parent  fd8d1a616be1c2ca1c1fc26a0be245eda6202059
configure.ac: Skip compiling JIT if LLVM_CONFIG is not specified

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -719,19 +719,16 @@ OCTAVE_CHECK_LIBRARY(z, ZLIB,
 dnl
 dnl
 dnl llvm is odd and has its own pkg-config like script. We should probably check
 dnl for existance and 
 dnl
 warn_llvm="LLVM library fails tests. JIT compilation will be disabled."
 
 AC_ARG_VAR(LLVM_CONFIG, [path to llvm-config utility])
-if test "x$ac_cv_env_LLVM_CONFIG_set" != "xset"; then
-   AC_PATH_TOOL([LLVM_CONFIG], [llvm-config])
-fi
 
 AC_ARG_ENABLE([jit-debug],
   AS_HELP_STRING([--enable-jit-debug], [Enable debug printing of jit IRs]))
 
 AS_IF([test "x$enable_jit_debug" = "xyes"], [
   AC_DEFINE(OCTAVE_JIT_DEBUG, 1, [Define for jit debug printing])
 ])
 
@@ -758,44 +755,49 @@ dnl We define some extra flags that llvm
 dnl Idealy we should get these from llvm-config, but llvm-config isn't very
 dnl helpful.
 dnl
 CPPFLAGS="-D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS $LLVM_CPPFLAGS $CPPFLAGS"
 CXXFLAGS="$LLVM_CXXFLAGS $CXXFLAGS"
 LIBS="$LLVM_LIBS $LIBS"
 LDFLAGS="$LLVM_LDFLAGS $LDFLAGS"
 
-AC_LANG_PUSH(C++)
-  AC_CHECK_HEADER([llvm/LLVMContext.h], [
-    AC_MSG_CHECKING([for llvm::getGlobalContext in llvm/LLVMContext.h])
-      AC_COMPILE_IFELSE(
-        [AC_LANG_PROGRAM([[#include <llvm/LLVMContext.h>]],
-                         [[llvm::LLVMContext& ctx = llvm::getGlobalContext ();]])],
-        [
-         AC_MSG_RESULT([yes])
-         warn_llvm=
-         XTRA_CXXFLAGS="$XTRA_CXXFLAGS $LLVM_CXXFLAGS $LLVM_CPPFLAGS"
-        ],
-        [AC_MSG_RESULT([no])
+if test "x$ac_cv_env_LLVM_CONFIG_set" = "xset"; then
+  AC_LANG_PUSH(C++)
+    AC_CHECK_HEADER([llvm/LLVMContext.h], [
+      AC_MSG_CHECKING([for llvm::getGlobalContext in llvm/LLVMContext.h])
+        AC_COMPILE_IFELSE(
+          [AC_LANG_PROGRAM([[#include <llvm/LLVMContext.h>]],
+                           [[llvm::LLVMContext& ctx = llvm::getGlobalContext ();]])],
+          [
+           AC_MSG_RESULT([yes])
+           warn_llvm=
+           XTRA_CXXFLAGS="$XTRA_CXXFLAGS $LLVM_CXXFLAGS $LLVM_CPPFLAGS"
+          ],
+          [AC_MSG_RESULT([no])
+      ])
     ])
-  ])
-AC_LANG_POP(C++)
+  AC_LANG_POP(C++)
+else
+  warn_llvm="LLVM_CONFIG not set. JIT compilation will be disabled."
+fi
+
 CPPFLAGS="$save_CPPFLAGS"
 CXXFLAGS="$save_CXXFLAGS"
 LIBS="$save_LIBS"
 LDFLAGS="$save_LDFLAGS"
 
 if test -z "$warn_llvm"; then
   AC_DEFINE(HAVE_LLVM, 1, [Define if LLVM is available])
 else
   LLVM_CXXFLAGS=
   LLVM_CPPFLAGS=
   LLVM_LDFLAGS=
   LLVM_LIBS=
-  AC_MSG_WARN([$warn_llvm])
+  OCTAVE_CONFIGURE_WARNING([warn_llvm])
 fi
 
 AC_SUBST(LLVM_CXXFLAGS)
 AC_SUBST(LLVM_CPPFLAGS)
 AC_SUBST(LLVM_LDFLAGS)
 AC_SUBST(LLVM_LIBS)
 
 ### Check for HDF5 library.
